### PR TITLE
Add shift() method

### PR DESCRIPTION
**Pull request by alex-verve**
refs #20

### CODE REVIEW

File: src/time_machine.py [hunk 1]
```diff
@@ -33,6 +34,19 @@ def time(self):
             _time_machine.original_time() - self.real_start_timestamp
         )
 
+    def tick(self, delta: dt.timedelta = DEFAULT_TICK_DELTA):
```
Comment (adamchainz) [inline src/time_machine.py:37]: no type hint, this is incorrect even

Also no need for a var, it's only used in one place

```suggestion
    def tick(self, delta=dt.timedelta(seconds=1)):
```

File: src/time_machine.py [hunk 2]
```diff
@@ -12,16 +12,17 @@
 import _time_machine
 
 NANOSECONDS_PER_SECOND = 1_000_000_000
+DEFAULT_TICK_DELTA = dt.timedelta(seconds=1)
```
Comment (adamchainz) [inline src/time_machine.py:15]: unnecessary
```suggestion
```

**Comment by adamchainz** [comment]
So I'm thinking this through a bit. I am not sure if it's wise to pursue 100% the same API as freezegun here because it's a bit confusing.

The name `tick` has two distinct meanings. The `tick` argument to `travel()` 'makes sense' to me - it "keeps the clock ticking." The `tick()` method - well it offsets the current time, even in reverse.

I'm also not sold on the default being 1 second. When is that useful case? Wouldn't it better to be explicit, for the sake of writing '1' in tests to make it clear how much time is moving by?

I'm gonna mull this over for a little bit, it's especially a question as to whether I want this library to have the exact same API as freezegun, or just to support the same use cases. There's some support for copying the core technique over to freezegun so I might be free-er to use a different API here.

**Comment by alex-verve** [comment]
> The tick() method - well it offsets the current time, even in reverse.

Totally agree, it is confusing to have two different things called `tick`.
I think that idea of having different API is wise in that case (as long as time-machine has covered all use cases).

What do you think of these?

```python
def test_coordinates_add_timedelta():
    with time_machine.travel(EPOCH, tick=False) as coordinates:
        coordinates += dt.timedelta(seconds=1)

        assert time.time() == EPOCH + 1


def test_coordinates_set_exact_time():
    destination_dt = dt.datetime(2000, 1, 1)

    with time_machine.travel(EPOCH, tick=False) as traveller:
        traveller.move_to(destination_dt)

        # or maybe even? :)
        traveller >> destination_dt

        assert time.time() == destination_dt.timestamp()
```

**Comment by adamchainz** [comment]
I don't like operator overloading, it is a bit implicit. Perhaps `shift()`? Or `move_offset()` ?

**Comment by alex-verve** [comment]
I like both `shift()` and `move_offset()` 😃 
Did you consider having one method for both things: `move(offset=..)/move(to=..)`? (not sure if it's good idea tho).
Maybe we need to hear opinion from thee wider public?

File: README.rst [hunk 3]
```diff
@@ -112,6 +112,29 @@ For example:
...
+
+.. code-block:: python
+
+    import datetime as dt
+    import time_machine
+
+    with time_machine.travel(0, tick=False) as traveller:
+        assert time.time() == 0
+
+        traveller.tick()
```
Comment (alex-verve) [inline README.rst:132]:
```suggestion
        traveller.shift()
```

File: README.rst [hunk 4]
```diff
@@ -112,6 +112,29 @@ For example:
...
+        assert time.time() == 1
+
+        traveller.tick(dt.timedelta(seconds=100))
```
Comment (alex-verve) [inline README.rst:135]:
```suggestion
        traveller.shift(dt.timedelta(seconds=100))
```

**Comment by adamchainz** [comment]
I've gone with `shift()` as you've seen. I think it's a good name, better than `tick()`. If anyone finds a problem with it, we can always add an alias.

File: README.rst [hunk 5]
```diff
@@ -112,6 +112,29 @@ For example:
...
+.. code-block:: python
+
+    import datetime as dt
+    import time_machine
+
+    with time_machine.travel(0, tick=False) as traveller:
+        assert time.time() == 0
+
+        traveller.shift()
+        assert time.time() == 1
```
Comment (alex-verve) [inline README.rst:133]: `delta` is obligatory now
```suggestion
```

File: README.rst [hunk 6]
```diff
@@ -112,6 +112,26 @@ For example:
...
+.. code-block:: python
+
+    import datetime as dt
+    import time_machine
+
+    with time_machine.travel(0, tick=False) as traveller:
+        assert time.time() == 0
+
+        traveller.shift(dt.timedelta(seconds=100))
+        assert time.time() == 101
```
Comment (alex-verve) [inline README.rst:133]: Sorry, I think this is the last one 😄 

```suggestion
        assert time.time() == 100
```

Comment (adamchainz) [reply README.rst:133]: I should really just read it shouldn't I? :P

**Pull request merged**